backend/src/v5/processors/teamspaces/projects/models/containers.js

```diff
@@ -72,4 +75,20 @@ Containers.deleteFavourites = async (username, teamspace, project, favouritesToR
 	return deleteFavourites(username, teamspace, accessibleContainers, favouritesToRemove);
 };
 
+Containers.updateSettings = async (teamspace, container, payload) => {
+	if (payload.defaultView) {
+		await checkViewExists(teamspace, container, payload.defaultView);
+	}
+
+	if (payload.defaultLegend) {
+		await checkLegendExists(teamspace, container, payload.defaultLegend);
+	}
+
+	const res = await updateModelSettings(teamspace, container, payload);
+
+	if (!res || res.matchedCount === 0) {
```

Comment:
do this test inside `updateModelSettings` 

Let the object closer to mongo to check mongo values.